src/de/grobox/liberario/fragments/DirectionsFragment.java

```diff
@@ -240,6 +244,23 @@ public void onResume() {
 		displayFavouriteTrips();
 		refreshAutocomplete(false);
 
+		// check if there's a preference for selected products for the current network
+		TransportNetwork current_network = Preferences.getTransportNetwork(getContext());
+		String network_pref = current_network.getId().toString() + "_selected_products";
+		SharedPreferences sp = getContext().getSharedPreferences(Preferences.NETWORK_SETTINGS, 0);
+		// get the selected products from the pref
+		Set<String> string_set = sp.getStringSet(network_pref, new HashSet<String>());
+        // if there is a pref for the current network, read the selected products and update
+        if(string_set.size()>0) {
+			EnumSet<Product> products = EnumSet.noneOf(Product.class);
+			for (String s : string_set) {
+				products.add(Product.valueOf(s));
+			}
+			Serializable tmpProducts = products;
+			products = (EnumSet<Product>) tmpProducts;
+			onProductsChanged(products);
```

Comment:
Why you do this in `onResume()` again after it has been done in `onCreateView()`? Any reason to not change the code in `onCreateView()` directly and initialize the products there?